Interval(oo, oo) returns S.EmptySet

arihantparsoya opened the pull request:
Fixes #11731. Added test for case when `Interval(oo, oo)` to return `S.EmptySet`.

### Sample Program
```python 
>>> from sympy import Interval, oo
>>> Interval(oo, oo)
EmptySet()
```


* Tests are added.

arihantparsoya committed:
Returns S.EmptySet for Interval(oo, oo)

Fixes #11731.

Added test case when `end == start` and `end == S.Infinity` to return
`S.EmptySet`.

Tests are added.

parsoyaarihant commented:
@asmeurer , can you review this?

asmeurer commented on sympy/sets/sets.py:
```diff
@@ -737,6 +737,8 @@ def __new__(cls, start, end, left_open=False, right_open=False):
         if end == start and (left_open or right_open):
             return S.EmptySet
         if end == start and not (left_open or right_open):
+            if start == S.Infinity:
```
Should work the same with negative infinity.

arihantparsoya committed:
Returns S.EmptySet for Interval(-oo, -oo)

Added test case when `end == start` and `end == S.NegativeInfinity` to
return
`S.EmptySet`.

parsoyaarihant commented:
@asmeurer , I have made the required changes.
